pkg/agent/agent.go

```diff
@@ -376,44 +316,67 @@ func (a *Agent) setAPIKeyConfigFrom(ctx context.Context, secrets []*kates.Secret
 // Watch is the work performed by the main goroutine for the Agent. It processes
 // Watt/Diag snapshots, reports to the Director, and executes directives from
 // the Director.
-func (a *Agent) Watch(ctx context.Context, snapshotURL, diagnosticsURL string) error {
+func (a *Agent) Watch(ctx context.Context) error {
 	dlog.Info(ctx, "Agent is running...")
-	ambHost, err := parseAmbassadorAdminHost(snapshotURL)
-	if err != nil {
-		// if we can't parse the host out of the url we won't be able to talk to ambassador
-		// anyway
-		return err
-	}
-
 	configCh := k8sapi.Subscribe(ctx, a.configWatchers.cond)
 	a.waitForAPIKey(ctx, configCh)
-
 	a.coreWatchers.EnsureStarted(ctx)
-	a.handleAmbassadorEndpointChange(ctx, ambHost)
+	a.handleAmbassadorEndpointChange(ctx, a.AESSnapshotURL.Hostname())
 	ambCh := k8sapi.Subscribe(ctx, a.ambassadorWatcher.cond)
 
-	// The following is kates that im not sure we can replicate with k8sapi as it currently exists
+	// The following is kates that I'm not sure if we can replicate with k8sapi as it currently exists
 	// leaving it in for now
-	client, err := kates.NewClient(kates.ClientConfig{})
+	//
+	_, resourcesLists, err := k8sapi.GetK8sInterface(ctx).Discovery().ServerGroupsAndResources()
 	if err != nil {
 		return err
 	}
-	ns := kates.NamespaceAll
-	dc := NewDynamicClient(client.DynamicInterface(), NewK8sInformer)
-	rolloutGvr, _ := schema.ParseResourceArg("rollouts.v1alpha1.argoproj.io")
-	rolloutCallback := dc.WatchGeneric(ctx, ns, rolloutGvr)
+	hasResource := func(r *schema.GroupVersionResource) bool {
```

Comment:
I am pretty sure that the cloud has argo-integrated features. If the user installs the agent, and then installs argo, the agent will not detect the argo resources, and the argo cloud features will not work. I suggest we leave it as it was for now, and build up k8sapi to monitor custom resources sometime in the future

Reply:
Leaving it as it is means that all telepresence users will see lots of error messages in the ambassador-agent log unless they install argo (which most telepresence users probably won't do). That in turn, will many of them to believe that all sorts of unrelated problems are related to those errors.

I'm not in favor of just leaving it as it is, and I think a solution where k8sapi is used will display the same problem (in the end, it's using the same RESTful API). Perhaps we need to check the resources periodically?

The simple solution is to document that if Argo is installed after the ambassador-agent, then the ambassador-agent needs to be restarted. What Helm-chart do we provide that adds those resources?